src/common/hooks/use-network-add-form.tsx

```diff
@@ -69,18 +81,10 @@ export const useNetworkAddForm = () => {
             if (!isLocal() && !values.url.includes('https://')) {
               _errors.url = 'The url needs to be https (non-local).';
             }
-            if (
-              networkList.find(item => item?.url?.split('//')?.[1] === values.url.split('//')[1])
-            ) {
-              _errors.general = 'This API has already been added.';
-            }
             try {
-              const _url = new URL(values.url);
-              const res = await fetchFromApi(`https://${_url.host}`)(DEFAULT_V2_INFO_ENDPOINT);
-              const data = await res.json();
-              data?.network_id && setNetworkMode(getChainIdFromInfo(data));
-              if (!data?.network_id) {
-                _errors.general = 'The API did not return a network_id.';
+              const networkId = await fetchNetworkId(values.url);
```

Comment:
Here we fetch the custom network info twice (for validation and for adding) which is not ideal. We could do more refactor here to only fetch it once, it might be ok though as it's happening rarely.